### PR TITLE
Combine api scrape dfs

### PR DESCRIPTION
**Pull request by andrew-curthoys**
added function to combine api_df and scrape_df dataframes

### CODE REVIEW

File: src/get_data.py [hunk 1]
```diff
@@ -194,12 +194,20 @@ def _get_players_on_ice(year : int, season : str, game_number : int) -> Tuple[st
     return response.text, year_id, game_id
 
 
-def _parse_players_on_ice(html : str, year_id : str, game_id : str) -> pd.DataFrame:
+def _parse_players_on_ice(html : str, year : int, game_id : str) -> pd.DataFrame:
```
Comment (yanniskatsaros) [inline src/get_data.py:197]: I like this, it's a cleaner approach.

File: src/get_data.py [hunk 2]
```diff
@@ -447,6 +457,31 @@ def get_roster(year : int, season : str, game_number : int) -> pd.DataFrame:
...
+
+def combine_api_scrape_data(api_df : pd.DataFrame, scrape_df : pd.DataFrame, year : int) -> pd.DataFrame:
+    """
+    Parameters 
+    ----------
+    api_df : DataFrame
+        A Pandas DataFrame constructed using the
+        _parse_api_plays() method
+
+    scrape_df : DataFrame
```
Comment (yanniskatsaros) [inline src/get_data.py:469]: It's typically convention to define functions/methods with `_function()` only if they are private and not public to the client. Is this the intended usage? Just double checking - if so, we would _typically_ not describe underlying implementation details on public functions, but only what they do. For a private function, it's ok to document internal API/developer level details.

**Review by yanniskatsaros — APPROVED** [review]
Apart from a possible documentation note/tweak, all looks good.

File: src/get_data.py (hunk 2 above)
Comment (andrew-curthoys) [reply src/get_data.py:469]: I imagine most of these functions will be private, and the public function would be a pipeline type process that calls all these functions in sequence & passes the proper variables between them. We will likely have to have a discussion on how we want to structure all this and if we want to make this a class that has class variables that are shared among methods as opposed to passing variables between the functions. I could be misunderstanding what you're talking about and I may not be making sense either, next time we get together we can hash it all out.

**Pull request merged**